index.js

```diff
@@ -17,6 +17,10 @@ function transformer(ast) {
   visit(ast, 'heading', visitor);
 
   function visitor(node) {
+    if (node.data && node.data.hProperties && 'id' in node.data.hProperties) {
+      slugs.slug(toString(node.data.hProperties.id), true);
```

Comment:
Probably want the second argument here to be an option. I'd imagine we want both usages of `slugs.slug()` to remain consistent.

Reply:
In this case the result of `slugs.slug` is not used... I added this to protect against duplicates...  but it won't help if the given id is already used.  Please see my other proposal in the comments.